### ISSUE TITLE
cisco.ios.ios_static_routes fails to collect facts with TypeError: must be str, not NoneType

### ISSUE THREAD

**Issue by michalstaporek** [ansible-collections/cisco.ios#178]
##### SUMMARY

The 'Gather listed static routes with provided configurations' fails on multiple devices with TypeError: must be str, not NoneType
Originally we thought it may be to do with Null0 summary routes, e.g. **ip route 172.20.56.0 255.255.255.0 Null0 name DMZ-SUMMARY** but I have found a few devices that have no Null routes but still give us errors.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

cisco.ios.ios_static_routes
##### ANSIBLE VERSION

```paste below
ansible 2.10.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/local/UK/admcmxs/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Sep 26 2019, 11:57:09) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### CONFIGURATION

```paste below
PERSISTENT_COMMAND_TIMEOUT(/etc/ansible/ansible.cfg) = 60
PERSISTENT_CONNECT_TIMEOUT(/etc/ansible/ansible.cfg) = 60
```

##### OS / ENVIRONMENT

jeshte-02a-sbc1 - WS-C3750X-48P running 15.2(4)E8
```
jeshte-02a-sbc1#sh running-config | section ^ip route|ipv6 route
ip route 10.72.30.128 255.255.255.192 10.72.28.10 tag 990 name JSY-CORP-WIFI
ip route 10.72.154.254 255.255.255.255 192.168.76.94
ip route 172.31.230.0 255.255.255.0 10.72.28.10 tag 990 name JSY-MOBILE-WIFI
ip route 192.168.72.70 255.255.255.255 10.72.29.10 tag 990 name jeshte-02a-vvg1_lo0
```
imdgoc-02a-sbc1 - C9300-24U running 16.8.1a
```
imdgoc-02a-sbc1#sh running-config | section ^ip route|ipv6 route
ip route 0.0.0.0 0.0.0.0 10.72.55.66 name default_route
ip route 10.72.53.64 255.255.255.192 10.72.55.66 tag 990 name Airmedia-DMZ
ip route 192.168.73.70 255.255.255.255 10.72.53.10 tag 990
```
##### STEPS TO REPRODUCE

run the example 'gathered' task from the documention

```yaml
  - name: Gather listed static routes with provided configurations
    cisco.ios.ios_static_routes:
      config:
      state: gathered
```

##### EXPECTED RESULTS

We should receive a list of static routes configured on the device

##### ACTUAL RESULTS

Playbook fails

```paste below
netcon3:UK\michalstaporek:~/code/inv-ansible-cisco> ansible-playbook -C -i hosts --ask-vault-pass static_routes.yaml --extra-vars "hosts_var=imdgoc-02a-sbc1" -vvv
ansible-playbook 2.10.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/local/UK/michalstaporek/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.6.8 (default, Sep 26 2019, 11:57:09) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
Using /etc/ansible/ansible.cfg as config file
Vault password: 
host_list declined parsing /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts as it did not pass its verify_file() method
script declined parsing /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts as it did not pass its verify_file() method
auto declined parsing /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts as it did not pass its verify_file() method
Parsed /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts inventory source with ini plugin
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios

PLAYBOOK: static_routes.yaml *******************************************************************************************************
1 plays in static_routes.yaml

PLAY [imdgoc-02a-sbc1] *************************************************************************************************************
META: ran handlers
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios

TASK [Gather listed static routes with provided configurations] ********************************************************************
task path: /home/local/UK/michalstaporek/code/inv-ansible-cisco/static_routes.yaml:25
redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
redirecting (type: terminal) ansible.builtin.ios to cisco.ios.ios
redirecting (type: cliconf) ansible.builtin.ios to cisco.ios.ios
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios
<192.168.73.60> ESTABLISH LOCAL CONNECTION FOR USER: UK\michalstaporek
<192.168.73.60> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk `"&& mkdir "` echo /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461 `" && echo ansible-tmp-1602668511.704204-103248-152980061479461="` echo /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461 `" ) && sleep 0'
<imdgoc-02a-sbc1> Attempting python interpreter discovery
<192.168.73.60> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<192.168.73.60> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
Using module file /home/local/UK/michalstaporek/.ansible/collections/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py
<192.168.73.60> PUT /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/tmp7p7dr2pf TO /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py
<192.168.73.60> EXEC /bin/sh -c 'chmod u+x /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/ /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py && sleep 0'
<192.168.73.60> EXEC /bin/sh -c '/usr/bin/python /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py && sleep 0'
<192.168.73.60> EXEC /bin/sh -c 'rm -f -r /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py", line 102, in <module>
    _ansiballz_main()
  File "/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_static_routes', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib64/python3.6/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py", line 692, in <module>
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py", line 687, in main
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py", line 91, in execute_module
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py", line 55, in get_static_routes_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/facts.py", line 114, in get_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py", line 131, in get_network_resources_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py", line 69, in populate_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py", line 146, in populate_destination
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py", line 105, in update_netmask_to_cidr
TypeError: must be str, not NoneType
fatal: [imdgoc-02a-sbc1]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103229althb9qk/ansible-tmp-1602668511.704204-103248-152980061479461/AnsiballZ_ios_static_routes.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_static_routes', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py\", line 692, in <module>\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py\", line 687, in main\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py\", line 91, in execute_module\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py\", line 55, in get_static_routes_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/facts.py\", line 114, in get_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py\", line 131, in get_network_resources_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py\", line 69, in populate_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py\", line 146, in populate_destination\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_e9_citec/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py\", line 105, in update_netmask_to_cidr\nTypeError: must be str, not NoneType\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP *************************************************************************************************************************
imdgoc-02a-sbc1            : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   


netcon3:UK\michalstaporek:~/code/inv-ansible-cisco> ansible-playbook -C -i hosts --ask-vault-pass static_routes.yaml --extra-vars "hosts_var=jeshte-02a-sbc1" -vvv
ansible-playbook 2.10.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/local/UK/michalstaporek/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.6.8 (default, Sep 26 2019, 11:57:09) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
Using /etc/ansible/ansible.cfg as config file
Vault password: 
host_list declined parsing /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts as it did not pass its verify_file() method
script declined parsing /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts as it did not pass its verify_file() method
auto declined parsing /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts as it did not pass its verify_file() method
Parsed /home/local/UK/michalstaporek/code/inv-ansible-cisco/hosts inventory source with ini plugin
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios

PLAYBOOK: static_routes.yaml *******************************************************************************************************
1 plays in static_routes.yaml

PLAY [jeshte-02a-sbc1] *************************************************************************************************************
META: ran handlers
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios

TASK [Gather listed static routes with provided configurations] ********************************************************************
task path: /home/local/UK/michalstaporek/code/inv-ansible-cisco/static_routes.yaml:25
redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
redirecting (type: terminal) ansible.builtin.ios to cisco.ios.ios
redirecting (type: cliconf) ansible.builtin.ios to cisco.ios.ios
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios
redirecting (type: action) cisco.ios.ios_static_routes to cisco.ios.ios
<192.168.72.60> ESTABLISH LOCAL CONNECTION FOR USER: UK\michalstaporek
<192.168.72.60> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo `"&& mkdir "` echo /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099 `" && echo ansible-tmp-1602668544.0200841-103372-231750393604099="` echo /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099 `" ) && sleep 0'
<jeshte-02a-sbc1> Attempting python interpreter discovery
<192.168.72.60> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<192.168.72.60> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
Using module file /home/local/UK/michalstaporek/.ansible/collections/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py
<192.168.72.60> PUT /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/tmpc3cvzq6v TO /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py
<192.168.72.60> EXEC /bin/sh -c 'chmod u+x /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/ /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py && sleep 0'
<192.168.72.60> EXEC /bin/sh -c '/usr/bin/python /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py && sleep 0'
<192.168.72.60> EXEC /bin/sh -c 'rm -f -r /home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py", line 102, in <module>
    _ansiballz_main()
  File "/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_static_routes', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib64/python3.6/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py", line 692, in <module>
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py", line 687, in main
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py", line 91, in execute_module
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py", line 55, in get_static_routes_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/facts.py", line 114, in get_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py", line 131, in get_network_resources_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py", line 69, in populate_facts
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py", line 146, in populate_destination
  File "/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py", line 105, in update_netmask_to_cidr
TypeError: must be str, not NoneType
fatal: [jeshte-02a-sbc1]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/local/UK/michalstaporek/.ansible/tmp/ansible-local-103349llwkocwo/ansible-tmp-1602668544.0200841-103372-231750393604099/AnsiballZ_ios_static_routes.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_static_routes', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py\", line 692, in <module>\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_static_routes.py\", line 687, in main\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py\", line 91, in execute_module\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/static_routes/static_routes.py\", line 55, in get_static_routes_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/facts.py\", line 114, in get_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py\", line 131, in get_network_resources_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py\", line 69, in populate_facts\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py\", line 146, in populate_destination\n  File \"/tmp/ansible_cisco.ios.ios_static_routes_payload_l9pq04ii/ansible_cisco.ios.ios_static_routes_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/static_routes/static_routes.py\", line 105, in update_netmask_to_cidr\nTypeError: must be str, not NoneType\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP *************************************************************************************************************************
jeshte-02a-sbc1            : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   


```

**Comment by justjais** [comment]
@michalstaporek Thanks for raising the issue and it should be fixed with the merge of PR: #[177](https://github.com/ansible-collections/cisco.ios/pull/177)

**Comment by justjais** [comment]
Closing the bug as the PR fixes the issue and is queued for IOS upcoming release

**Comment by michalstaporek** [comment]
Hello @justjais 
Thank you very much for an update, I've tested the fix from PR #177 and it does indeed resolve the issue.
